Shadcn sidebar

rohitverma007 opened the pull request:
Ready for review.

Migrated the existing navbar to shadcn sidebar component. 

Here's a quick demo video:

https://www.loom.com/share/a36f71ddd540476bbf447efcc4bad8ca?sid=9bf794ec-e8f5-4e91-81fd-7cb65f414fc6

Here are some screenshots:
<details>
<summary>Desktop Screnshots</summary>
<img width="1459" alt="Screenshot 2025-04-14 at 5 19 59 PM" src="https://github.com/user-attachments/assets/034b85c0-bdc3-410f-be33-b0f1e20e3341" />


<details>
<summary>Mainly would like to get feedback on the company switching logic and if this looks alright:</summary>
<img width="1461" alt="Screenshot 2025-04-14 at 5 29 55 PM" src="https://github.com/user-attachments/assets/b2a284cf-a6d2-4705-af32-56d22f478357" />
<img width="1426" alt="Screenshot 2025-04-14 at 5 41 58 PM" src="https://github.com/user-attachments/assets/6c9f2150-27be-4eb4-9492-f0472cfdf34d" />
</details>

~~The circle notification icon beside company name in dropdown is potentially misleading it's simply meant to be an indication of which company is currently selected. Perhaps a better indication would be to grey out the selected company. ~~ Leaving the default circle icon for now.
</details>

<details>

<summary>The mobile sidebar is a bit different than previous one:</summary
<img width="341" alt="Screenshot 2025-04-17 at 3 29 53 PM" src="https://github.com/user-attachments/assets/0216dcd9-9ce6-4f68-ac29-2c519d60caa8" />
<img width="303" alt="Screenshot 2025-04-17 at 3 29 44 PM" src="https://github.com/user-attachments/assets/dcda8115-9b07-4310-991c-5032c69e7623" />
<img width="308" alt="Screenshot 2025-04-17 at 3 29 40 PM" src="https://github.com/user-attachments/assets/40542cb2-5978-4636-997e-90ef3af3e800" />

</details>

Todos:

- [x] figure out + test all "company" scenarios - none , 1 , 1+, implement edge cases
- [x] figure out + test all "roles" + implement edge case
- [x] add mobile shadcn sidebar
- [x] refactor SidebarMenuItem (DRYify)
- [ ] fix css to match old nav [leaving this for a different (internal?) PR/ design review ]:
  - [ ]     fix color of default + selected nav item text (match the old navbar)
  - [ ]     fix account icon (default + selected)

part of #18 (and eventually #17)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/antiwork/flexile?pullRequest=126) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/antiwork/flexile?pullRequest=126) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/antiwork/flexile?pullRequest=126) it.</sub>

slavingia commented:
Grey out would work. X uses a checkmark, which may work too. But imo it's not bad atm and we can tweak in a future design-minded PR

![image](https://github.com/user-attachments/assets/0022e053-b63d-46be-9928-1b38d4baa7b2)

slavingia commented:
I think going for pixel parity isn't the goal here btw. Shadcn makes some nice improvements to modernize the look, which I tried to copy here - https://github.com/antiwork/flexile/issues/17

But also something we can iterate on internally

rohitverma007 commented:
> I think going for pixel parity isn't the goal here btw. Shadcn makes some nice improvements to modernize the look, which I tried to copy here - #17
> 
> But also something we can iterate on internally

Thanks for the feedback, good to know, will focus less on pixel perfectness. Btw, images on #17 are broken if you are referring to those, can still access v0 fwiw.

slavingia commented:
Updated them

rohitverma007 commented:
@slavingia wrapped up the remaining items and added a short loom demo video. The PR is ready for review by the team :)
I believe I’ve addressed everything and confident with the testing I have done, but since this touches the app’s main entrypoint, would be worthwhile to give it a thorough review. Thanks!

slavingia commented on apps/next/components/layouts/Main.tsx:
```diff
@@ -131,206 +150,370 @@ export default function MainLayout({
+                  isActive={pathname.startsWith("/settings")}
+                  label="Account"
+                />
+                {/* <SidebarMenuItem>
```
?

rohitverma007 replied:
removed 😅

slavingia commented:
There seem to be other diffs unrelated to this. I've run autofix and the Vercel deploy. Please self-review and then I'll have someone internally do a pass and merge.

rohitverma007 commented:
> There seem to be other diffs unrelated to this. I've run autofix and the Vercel deploy. Please self-review and then I'll have someone internally do a pass and merge.

Indeed, updated to remove the unnecessary changes.

raulpopadineti commented on apps/next/components/layouts/Main.tsx:
```diff
@@ -364,225 +539,44 @@ const useSwitchCompanyOrRole = () => {
   };
 };
 
-const NavLinks = ({ company }: { company: Company }) => {
```
Why couldn't we have reused this since it's scoped per `company` and makes rendering sidebar nav items cleaner instead of always having to pass the `openCompany.routes.some(route => route.label === label)` for each, like it is now?

Right now it looks like a big diff and hard to determine if something else changed besides replacing the nav item components with Shadcn's.

rohitverma007 replied:
Ah good point, when I started building it out, it made sense to rewrite all of that but I see the benefit of re-using the NavLinks. Can refactor this, although won't be able to get to it until around the end of the weekend.

slavingia replied:
Sounds good. Please have done by end of weekend so we can merge in around Noon ET latest on Monday.

rohitverma007 replied:
@raulpopadineti I have updated the code to re-use the NavLinks and NavLink component which makes the diff more readable. Feel free to give it a look. Reverting back to these components also fixed the other updates href bug. Let me know if there are any other changes I should make, thanks!

slavingia commented:
Great. Turned out to be smaller change than expected which is great. We can merge it in today!

rohitverma007 commented on apps/next/components/layouts/Main.tsx:
```diff
@@ -131,206 +150,362 @@ export default function MainLayout({
+                  {openCompany.routes.some(route => route.label === "Updates") && (
+                    <>
+                      <SidebarNavItem
+                        href="/updates"
```
Updating to `NavLinks` and `NavLink` fixes this.

slavingia commented:
Doesn't meet our quality bar. We'll do this internally as it's a small lift.

rohitverma007 commented:
> Doesn't meet our quality bar. We'll do this internally as it's a small lift.

@slavingia For my own understanding and to help me reflect and improve going forward, do you mean the quality in terms of code or the ui/ux? After the re-use of NavLinks and NavLink it seemed like a fairly minor diff and straight forward, any further feedback would be greatly appreciated.

slavingia commented on apps/next/components/layouts/Main.tsx:
```diff
@@ -36,26 +37,48 @@ import { useQuery, useQueryClient } from "@tanstack/react-query";
 import { company_search_path, company_switch_path } from "@/utils/routes";
 import { formatDate } from "@/utils/time";
 import { useOnGlobalEvent } from "@/utils/useOnGlobalEvent";
+import "@/components/layouts/sidebar.css";
```
?

rohitverma007 replied:
oops, removed.

slavingia commented:
Sorry for closing the PR with a terse comment. If you want to pair with @MayaRainer or me on re-doing this and merging this in, that may be a better path forward. Just reach out to maya@gumroad.com / sahil@gumroad.com to schedule a call. It's quite urgent so in the next day would be lovely!

rohitverma007 commented:
> Sorry for closing the PR with a terse comment. If you want to pair with @MayaRainer or me on re-doing this and merging this in, that may be a better path forward. Just reach out to [maya@gumroad.com](mailto:maya@gumroad.com) / [sahil@gumroad.com](mailto:sahil@gumroad.com) to schedule a call. It's quite urgent so in the next day would be lovely!

Thanks for this, appreciate the time and help, have reached out to @MayaRainer with some times over the next day. Looking forward to it.

slavingia commented:
Why isn't this merged?

MayaRainer commented:
Was on a call with @rohitverma007 and we discussed some changes that had to be made to fix tests and some broken layout. Should be ready for re-review by tomorrow.

rohitverma007 commented:
fixed some things from the call, namely lint issues + using shadcn sidebar for "updates" nav item 
still working through the mobile / defaulting to shadcn sidebar trigger, need a bit more time to wrap that up, should be ready for re-review within a day.

MayaRainer commented:
Thanks for working on this! Due to the urgency I'm going to bring this over the finish line today, but still feel free to reach out about the bounty when they're implemented.

rohitverma007 commented:
> Thanks for working on this! Due to the urgency I'm going to bring this over the finish line today, but still feel free to reach out about the bounty when they're implemented.

Sure yea, go ahead. Thanks for your help and time. Looking forward to contributing to other issues.

slavingia commented:
Using a new PR to merge these changes!
